functions/src/resolve-market.ts

```diff
@@ -74,6 +78,20 @@ export const resolvemarket = newEndpoint(opts, async (req, auth) => {
   const creator = await getUser(creatorId)
   if (!creator) throw new APIError(500, 'Creator not found')
 
+  if (contract.outcomeType === 'FREE_RESPONSE') {
+    if (resolutions) {
+      Object.keys(resolutions).forEach((outcome) =>
+        validateFreeResponseOutcome(contract, outcome)
+      )
+      const pctSum = sum(Object.values(resolutions))
+      if (pctSum !== 100) {
+        throw new APIError(400, 'Resolution percentages must sum to 100')
+      }
+    } else if (!isNaN(+outcome)) {
```

Comment:
`z.number` already excludes `NaN`.

Reply:
This was intended to check "is outcome a number". Switched to using lodash's `isNumber`.